core/ibft.go

```diff
@@ -133,18 +134,26 @@ func NewIBFT(
 	}
 }
 
+// SetTime function set duration to gauge
+func SetTime(prefix string, startTime time.Time) {
+	metrics.SetGauge([]string{"go-ibft", prefix, "duration"}, float32(time.Since(startTime).Seconds()))
+}
+
 // startRoundTimer starts the exponential round timer, based on the
 // passed in round number
 func (i *IBFT) startRoundTimer(ctx context.Context, round uint64) {
 	defer i.wg.Done()
 
+	startTime := time.Now()
+
 	roundTimeout := getRoundTimeout(i.baseRoundTimeout, i.additionalTimeout, round)
 
 	//	Create a new timer instance
 	timer := time.NewTimer(roundTimeout)
 
 	select {
 	case <-ctx.Done():
+		SetTime("round", startTime)
```

Comment:
Do we need to call `SetTime` in `startRoundTimer`? I think that for each `RunSequence` we will have two `metrics.SetGauge([]string{"go-ibft", prefix, "duration"}, float32(time.Since(startTime).Seconds()))`. And I am not sure if we want that

Reply:
Not sure what you mean. We want to measure round and sequence. That is why we measure time for `RunSequence` and `startRoundTimer`